ws-client/src/manager.rs

```diff
@@ -34,32 +34,60 @@ pub enum RequestStatus {
 	Invalid,
 }
 
-type PendingCallOneshot = oneshot::Sender<Result<JsonValue, Error>>;
+type PendingCallOneshot = Option<oneshot::Sender<Result<JsonValue, Error>>>;
 type PendingSubscriptionOneshot = oneshot::Sender<Result<(mpsc::Receiver<JsonValue>, SubscriptionId), Error>>;
 type SubscriptionSink = mpsc::Sender<JsonValue>;
 type UnsubscribeMethod = String;
-type RequestId = u64;
+type RequestId = u8;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 /// Manages and monitors JSONRPC v2 method calls and subscriptions.
 pub struct RequestManager {
+	/// Free list, 0 indicates a free slot and 1 an occupied slot.
+	free_list: [u8; 256],
```

Comment:
we could make this faster by reqister the last fetched request ID instead of starting from zero every time in `fn next_request_id`

Reply:
after some thinking, it's probably unnecessary to make assumptions about how many unanswered requests which covers all use-cases.